README.md

```diff
@@ -308,6 +476,35 @@ Options:
 * `"fileExtension"`: An optional extension to add to resulting files.
 * `"iriToPath"`: A collection of mappings that indicate what URL patterns should be translated into what folder structure.
 
+#### Annotated File Quad Sink
+
+A quad sink that writes to files using an IRI to local file system path mapping with an annotation in each file in the form of triples.
```

Comment:
Can't this be achieved using one of the quad transformers?
I believe we similarly use them for generating the links to type indexes.

Reply:
I've opted for this solution because, with the quads, I cannot know in which file the data is going to go hence if we have a fragmentation that puts multiple entities (for instance posts) in the same file there are going to be n (number of posts) time the annotation. My intention is really to add the annotation in each file and the file structure is somewhat independent from the knowledge graph.

Reply:
It could be done that way, but I am not sure if the state management is would be worth it. I also think that this sink could used for other purposes. But if it is wrong I can create a new transformer or figure out how to do that with the current ones.

Reply:
The main problem of this approach is that it does not align with the overal idea of the architecture.
We have quad sources -> transformers -> fragmenters -> sinks, each having their precise goals.
But by adding this type of sink, we move into the territory of transformers, without having the composability functionalities that transformers have.
Aside from that, it also adds confusion for users of this package, since there are now two different ways to achieve the same thing.
So I would favor an approach that is more aligned with the overal architecture.
And we already have a similar approach for the type index, so it's definitely possible.

> If we don't care about duplicates then it would be a different matter, however, I think this quad sink is closer to the intention.

This one can help: https://github.com/SolidBench/rdf-dataset-fragmenter.js/?tab=readme-ov-file#distinct-quad-transformer

Reply:
I will implement it that way thank you!